### ISSUE TITLE
Switching between List view and Calendar View not working

### ISSUE THREAD

**Issue by jigneshpshah** [frappe/erpnext#15517]
Environment 
ERPNext: v10.1.52 (master)
Frappe Framework: v10.1.47 (master)

After switching 2-3 times between List View and Calendar View it stops and only List view is shown as below.

Tested on Timesheet doctype.

**Actual Behavior**

![calendarview error](https://user-images.githubusercontent.com/4900815/46135299-79bfda00-c262-11e8-8c36-86fca16658a0.gif)

**Expected Behavior**

Switch between different views should happen seamlessly.

**Comment by crescent28** [comment]
It only seems that Calendar option has to be clicked twice to switch after list. 
![timesheet](https://user-images.githubusercontent.com/41110073/46273386-edb7f600-c572-11e8-8ef1-038584540a0e.gif)

**Comment by joolsr** [comment]
Yes I see this too in current master branch. The good news is that it appears this is fixed in my testing of V11 though 😀
